qpc/insights/upload.py

```diff
@@ -147,13 +183,51 @@ def _validate_args(self):
                      self.report_id)))
         print(_(messages.INSIGHTS_RETRIEVE_REPORT % self.report_id))
 
+    def verify_report_details(self):
+        """
+        Verify that the report contents are a valid deployments report.
+
+        :returns boolean regarding report validity
+        """
+        # validate required keys
+        report_contents = extract_json_from_tar(self.response.content,
+                                                print_pretty=False)
+        required_keys = ['report_platform_id',
+                         'report_id',
+                         'report_version',
+                         'report_type',
+                         'system_fingerprints']
+        for key in required_keys:
+            present = report_contents.get(key)
```

Comment:
Let's gather all the keys that are missing and generate a message that lists all of them.